src/vm_stack.zig

```diff
@@ -343,7 +343,7 @@ const Stack = struct {
 
         const returns_source: []const Val = stack.values[source_begin..source_end];
         const returns_dest: []Val = stack.values[dest_begin..dest_end];
-        std.mem.copy(Val, returns_dest, returns_source);
+        std.mem.copyForwards(Val, returns_dest, returns_source);
```

Comment:
`@memcpy` + stack overlap assert?

Reply:
Added the assert, but `@memcpy` doesn't work here.